dcos/dcos-ui
# DCOS-51000 - Group aggregate status value

Introduced the `ServiceTreeStatusSummary` response type. 
This summary object can be used to render a status
that can also be translated using the id & value key values. It also
contains the highest priority statuses display name and a count of
statuses grouped by the StatusIcon.name. The highest priority status
count is used for the displayed status values, but all are returned so
they can be used to generate the desired tooltip content as well.

- Closes DCOS-51000
- Closes DCOS-50999

## Testing


1. Navigate to the Service page
2. Create a group
3. Inside your group create 2+ services / pods / groups
4. Navigate back to Services overview
5 Verify status is an aggregate of child service status with the highest priority showing.

## Trade-offs

- I introduced a `name` key to the `StatusIcon` type which required some small refactoring. I used this when generating the aggregate status counts for the `ServiceTreeStatusSummary`.
- I don't love the naming of the keys on `ServiceTreeStatusSummary`, open to better suggestions here. 

## Dependencies

This is based on #3748 and should be merged after it, I have the base set to the branch for now so the diff is reviewable. But that mean CI is broken until #3748 is merged to `master` and this PRs base can be updated.

## Screenshots

<img width="1145" alt="image" src="https://user-images.githubusercontent.com/1972968/55591016-1bd0eb00-56fa-11e9-8c89-e75e0616e6ab.png">
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/1972968/55591043-2d19f780-56fa-11e9-955e-cc3042415b26.png">
